app/services/case_creator.rb

```diff
@@ -0,0 +1,33 @@
+class CaseCreator
+  attr_reader :tribunal_case, :errors
+  attr_reader :payment_url
+
+  def initialize(tribunal_case)
+    @tribunal_case = tribunal_case
+    @errors = []
+  end
+
+  def call
+    begin
+      glimr_case = GlimrNewCase.new(tribunal_case).call!
+
+      payment = PaymentUrl.new(
+          case_reference: glimr_case.case_reference,
+          confirmation_code: glimr_case.confirmation_code).call!
+
+      @payment_url = payment.payment_url
+    rescue => ex
+      errors << ex.message
```

Comment:
Something for later maybe, but should we think about how/if we expose these errors to the user?